exchange/wrappers/luno.js

```diff
@@ -51,20 +67,45 @@ const recoverableErrors = [
   'NOTFOUND'
 ];
 
+
 const processResponse = function(funcName, callback) {
   return (error, body) => {
+    /* BitX Error Codes
+          Error: BitX error 400: invalid id
+          Error: BitX error 401: "error":"API key not found","error_code":"ErrAPIKeyNotFound","error_action":
+          Error: BitX error 429: -- API limit
+          Error: BitX error 500: Something went wrong, we're looking into it.
+     */
+
     if (!error && !body) {
-      error = new Error('Empty response');
+      error = new Error('received empty response');
     }
 
     if (error) {
+      log(name, funcName + '() received error -->', error.message);
+
       if (includes(error.message, recoverableErrors)) {
         error.notFatal = true;
       }
 
       if (includes(error.message, ['error 429'])) {
         error.notFatal = true;
-        error.backoffDelay = 10000;
+        error.backoffDelay = 3000;
+      }
+
+      if (includes(error.message, ['error 500'])) {
+        error.notFatal = true;
+        error.backoffDelay = 15000;
+      }
+
+      if (includes(error.message, ['Insufficient balance'])) {
+        error.notFatal = true;
+        error.backoffDelay = 2000;
+        tradeAttempt++;
```

Comment:
this is okay for now, but we should aim to avoid these kind of things in the future!